consensus/bor/bor.go

```diff
@@ -1170,44 +1156,50 @@ func (c *Bor) GetPendingStateProposals(snapshotNumber uint64) ([]*big.Int, error
 func (c *Bor) CommitStates(
 	state *state.StateDB,
 	header *types.Header,
-	chain core.ChainContext,
+	chCtx chainContext,
 ) error {
-	// get pending state proposals
-	stateIds, err := c.GetPendingStateProposals(header.Number.Uint64() - 1)
-	if err != nil {
-		return err
-	}
-
-	// state ids
-	if len(stateIds) > 0 {
-		log.Debug("Found new proposed states", "numberOfStates", len(stateIds))
-	}
-
-	method := "commitState"
-
-	// itereate through state ids
-	for _, stateID := range stateIds {
-		// fetch from heimdall
-		response, err := c.HeimdallClient.FetchWithRetry("clerk", "event-record", strconv.FormatUint(stateID.Uint64(), 10))
+	number := header.Number.Uint64()
+	if number < c.config.Sprint {
+		return errors.New("Requested to commit states too soon")
+	}
+
+	from := c.EventsLastFetchededAt
```

Comment:
`c.EventLastFetchedAt` can be empty (zero) when you restart the node.

Reply:
You are also not updated `EventsLastFetchededAt` anywhere.

Reply:
Using stateReceiver contract instead.